sysbuild/CMakeLists.txt

```diff
@@ -506,19 +506,20 @@ function(${SYSBUILD_CURRENT_MODULE_NAME}_pre_cmake)
 
   if(SB_CONFIG_BT_FAST_PAIR)
     set_config_bool(${DEFAULT_IMAGE} CONFIG_BT_FAST_PAIR y)
+  else()
+    set_config_bool(${DEFAULT_IMAGE} CONFIG_BT_FAST_PAIR n)
+  endif()
 
-    if(DEFINED FP_MODEL_ID AND DEFINED FP_ANTI_SPOOFING_KEY)
-      if(SB_CONFIG_PARTITION_MANAGER)
-        include(${ZEPHYR_NRF_MODULE_DIR}/cmake/sysbuild/fast_pair_hex.cmake)
-      endif()
-
-      set(FP_DATA_PRESENT "y" CACHE INTERNAL "Fast Pair provisioning data provided" FORCE)
-    else()
-      message(WARNING "Fast Pair support is enabled but `FP_MODEL_ID` or `FP_ANTI_SPOOFING_KEY` were not provided, this is likely to cause a build error")
-      set(FP_DATA_PRESENT "n" CACHE INTERNAL "Fast Pair provisioning data provided" FORCE)
+  if(SB_CONFIG_BT_FAST_PAIR_PROVISIONING_DATA_GENERATE)
+    if(SB_CONFIG_PARTITION_MANAGER)
+      include(${ZEPHYR_NRF_MODULE_DIR}/cmake/sysbuild/fast_pair_hex.cmake)
     endif()
   else()
-    set_config_bool(${DEFAULT_IMAGE} CONFIG_BT_FAST_PAIR n)
+    if(SB_CONFIG_BT_FAST_PAIR)
+      message(WARNING "Fast Pair support is enabled but automatic Fast Pair provisioning data generation is disabled. "
```

Comment:
check if either value was provided and emit the warning if so otherwise do not emit anything